lib/vagrant-cloudstack/action/run_instance.rb

```diff
@@ -251,6 +241,34 @@ def call(env)
           @app.call(env)
         end
 
+        def store_password(env, domain_config, job_id)
+          server_job_result = env[:cloudstack_compute].query_async_job_result({:jobid => job_id})
+          if server_job_result.nil?
+            env[:ui].warn(' -- Failed to retrieve job_result for retrieving the password')
+            return
+          end
+
+          while true
```

Comment:
This kind of poling logic could be better implemented using the `Util::Timer` class. Please have a look at line 160 from`run_instance.rb`. Can you use the same approach here?